services/continuous_querier/config.go

```diff
@@ -24,13 +25,23 @@ type Config struct {
 	// every minute, this should be set to 1 minute. The default is set to '1s' so the interval
 	// is compatible with most aggregations.
 	RunInterval toml.Duration `toml:"run-interval"`
+
+	// CQ timeout for how long to wait for the previous query to complete, the default is `1h` so
+	// the timeout is compatible with most CQs that are run.
+	CQTimeout toml.Duration `toml:"query-timeout"`
+
+	// CQ denominator of how many (out of the total CQs configured) to run concurrently
+	// default: 2 (50%)
+	CQDenominator int `toml:"query-denominator"`
 }
 
 // NewConfig returns a new instance of Config with defaults.
 func NewConfig() Config {
 	return Config{
-		LogEnabled:  true,
-		Enabled:     true,
-		RunInterval: toml.Duration(DefaultRunInterval),
+		LogEnabled:    true,
+		Enabled:       true,
+		RunInterval:   toml.Duration(DefaultRunInterval),
+		CQTimeout:     toml.Duration(DefaultCQTimeout),
```

Comment:
I'm not sure a CQ timeout should be included at all. If one is included, I think it should possibly be done based on the GROUP BY or the RESAMPLE FOR intervals rather than a static value that's true for all CQs. The primary reason I would be hesitant to include one at all is just because we already apply query limits within the coordinator component for queries. One thing that probably should be included though is a warning or error message if we try to run a CQ and the CQ pipeline is blocked so an administrator can come around and unblock the continuous query pipeline.